tools/sgcollect_info

```diff
@@ -385,9 +395,22 @@ def make_collect_logs_tasks(
         os_home_dirs.append(guessed_log_path)
 
     # Keep a dictionary of log file paths we've added, to avoid adding duplicates
-    sg_log_file_paths = {}
-
-    sg_tasks = []
+    sg_log_file_paths: set[pathlib.Path] = set()
+
+    sg_tasks: List[PythonTask] = []
+
+    def add_file_collection_task(filename: pathlib.Path):
+        canonical_filename = pathlib.Path(filename)
+        canonical_filename.resolve()
+        if canonical_filename in sg_log_file_paths:
+            return
+        sg_log_file_paths.add(canonical_filename)
+        if not should_redact or not redactable_file(filename):
+            task = add_file_task(sourcefile_path=filename)
```

Comment:
I wonder if we should use the canonicalised version of the filename here or not? It won't make much material difference other than the path that gets included in the file header I guess

Reply:
These are written without headers, so I am not sure this is helpful. I am not opposed to writing them with headers.